cmake/checks/check_01_cxx_features.cmake

```diff
@@ -122,10 +118,26 @@ IF(NOT DEFINED DEAL_II_WITH_CXX14 OR DEAL_II_WITH_CXX14)
       "
       DEAL_II_HAVE_CXX14_MAKE_UNIQUE)
 
+    #
+    # Clang-3.5* or older, bail out with a spurious error message in case
+    # of an undeduced auto return type.
+    #
+    PUSH_CMAKE_REQUIRED("${DEAL_II_CXX_FLAGS_DEBUG}")
```

Comment:
A year or two from now, we will scratch our heads what exactly the problem was. Can you add a link to the stackexchange site that showed the bug?